pkg/api/handlers/libpod/images.go

```diff
@@ -498,12 +505,61 @@ func CommitContainer(w http.ResponseWriter, r *http.Request) {
 	if len(query.Repo) > 0 {
 		destImage = fmt.Sprintf("%s:%s", query.Repo, tag)
 	}
-	commitImage, err := ctr.Commit(r.Context(), destImage, options)
+
+	var commitImage *libimage.Image
+
+	success := false
+	runCtx, cancel := context.WithCancel(r.Context())
+	go func() {
+		defer cancel()
+		commitImage, err = ctr.Commit(r.Context(), destImage, options)
+		if err == nil {
+			success = true
+		} else {
+			stdout.Write([]byte(err.Error() + "\n"))
+		}
+	}()
+
 	if err != nil && !strings.Contains(err.Error(), "is not running") {
 		utils.Error(w, http.StatusInternalServerError, fmt.Errorf("CommitFailure: %w", err))
 		return
 	}
-	utils.WriteResponse(w, http.StatusOK, entities.IDResponse{ID: commitImage.ID()})
+	flush := func() {
```

Comment:
This change will break existing users.  So we need to add a new parameter to enable the new stream format etc.

Reply:
How about I add a stream option to the bindings and then just return the old function when not asking for streaming.

Reply:
I didn't think about the bindings yet but they may need some massaging as well.

My comment here only addressed the REST API.  In this case, we need a new parameter to flip the stream to avoid breaking existing users (including podman-py).

Reply:
I changed the code to use Streaming flag, in the case of `podman --remote commit` this defaults to True, all other APIs will default to False and should fall back to old behaviour.

If the user calls podman --remote commit --quiet ...

It turns the Stream flag to false, and falls back to the old behaviour.